File: src/main/java/ru/javarush/cryptoanalyzer/konovalov/Commands/CaesarDecoder.java
package ru.javarush.cryptoanalyzer.konovalov.Commands;

import ru.javarush.cryptoanalyzer.konovalov.entity.Result;

public class CaesarDecoder implements CaesarAction {

    @Override
    public Result execute(String[] parameters) {
        String encryptedText = parameters[0];
        String decryptedText = parameters[1];
        int key = Integer.parseInt(parameters[2]);

        return applyCipherKey(encryptedText,decryptedText,key*(-1));

    }

}

File: src/main/kotlin/commands/CaesarDecoder.kt
package commands
import entity.Result


class CaesarDecoder:  CaesarAction {
    override fun execute(parameters: Array<String>): Result {
        val encryptedText : String = parameters[0]
        val decryptedText : String = parameters[1]
        val key: Int = parameters[2].toInt()
        val invertedKey = key * (-1)

        return applyCipherKey(encryptedText,decryptedText,invertedKey)
    }
}


